### PR TITLE
Fix eslint errors in create-config/index.js

### PR DESCRIPTION
**Pull request by mreishus**
#### Changes proposed in this Pull Request

* Fix eslint errors causing pipeline failures in any PRs touching `client/lib/create-config/index.js`

#### Testing instructions

* `yarn run eslint client/lib/create-config/index.js`

### CODE REVIEW

File: client/lib/create-config/index.js [hunk 1]
```diff
@@ -58,10 +58,9 @@ const config = ( data ) => ( key ) => {
 /**
  * Checks whether a specific feature is enabled.
  *
- * @param {string} feature Feature name
  * @param {object} data the json environment configuration to use for getting config values
+ * curried param {string} feature Feature name
  * @returns {boolean} True when feature is enabled.
- * @api public
```
Comment (cpapazoglou) [inline client/lib/create-config/index.js:64]: Any info on why this is good to be removed (apart from that it raised errors)?

File: client/lib/create-config/index.js [hunk 2]
```diff
@@ -58,10 +58,9 @@ const config = ( data ) => ( key ) => {
 /**
  * Checks whether a specific feature is enabled.
  *
- * @param {string} feature Feature name
  * @param {object} data the json environment configuration to use for getting config values
+ * curried param {string} feature Feature name
```
Comment (cpapazoglou) [inline client/lib/create-config/index.js]: Nice workaround!

**Review by cpapazoglou — APPROVED** [review]
Thanks @mreishus! Generally this works well, I am just worried about the 
`@api public` removal. It seems like a commonly used annotation. Can't find any documentation though.

File: client/lib/create-config/index.js (hunk 1 above)
Comment (mreishus) [reply client/lib/create-config/index.js:64]: Hmm, no, just trying to get rid of the error. I can re-add it without the `@`, which fixes the error but keeps the info? e88ef3c6bacd404

Comment (cpapazoglou) [reply client/lib/create-config/index.js:64]: I posted a question here p1605896174269200-slack-C7YPUHBB2

Comment (mreishus) [reply client/lib/create-config/index.js:64]: Ok, I moved to `@public`, which is the same as `@access public`.  Seems to convey the same information as `@api public` while passing eslint and meeting the JSdoc spec.

**Comment by mreishus** [comment]
Updated the format of comment for curried function after reading [a github thread](https://github.com/jsdoc/jsdoc/issues/1286) about it.  This seems to work in my editor:

**Pull request merged**